### PR TITLE
Arsenal framework doc

### PR DESCRIPTION
**Pull request by alganthe**
**When merged this pull request will:**
- Add a wiki page for the ACE arsenal framework 

Note: I couldn't live test the page this time.

### CODE REVIEW

File: docs/wiki/framework/arsenal-framework.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+---
+layout: wiki
+title: ACE arsenal framework
+description: Explains how to interact with the ACE arsenal APIs
```
Comment (jonpas) [inline docs/wiki/framework/arsenal-framework.md]: Missing final dot, `Arsenal`, `API` singular.

File: docs/wiki/framework/arsenal-framework.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
+---
+layout: wiki
+title: ACE arsenal framework
```
Comment (jonpas) [inline docs/wiki/framework/arsenal-framework.md]: `ACE Arsenal Framework`

File: docs/wiki/framework/arsenal-framework.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+parent: wiki
+mod: ace
+version:
+  major: 3
+  minor: 12
+  patch: 0
+---
+
+<div class="panel callout">
+    <h5>BI functions and variables from BI arsenal DO NOT apply to ACE arsenal!</h5>
```
Comment (jonpas) [inline docs/wiki/framework/arsenal-framework.md]: `from BI Virtual Arsenal`

File: docs/wiki/framework/arsenal-framework.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+</div>
+
+<div class="panel callout">
+    <h5>Note:</h5>
+    <p>Units and objects that can have ACE arsenal added to them will be called "Boxes" in this documentation</p>
```
Comment (jonpas) [inline docs/wiki/framework/arsenal-framework.md]: Missing final dot, `"boxes"`.

File: docs/wiki/framework/arsenal-framework.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+<div class="panel callout">
+    <h5>BI functions and variables from BI Virtual Arsenal DO NOT apply to ACE Arsenal!</h5>
+</div>
+
+<div class="panel callout">
+    <h5>Note:</h5>
+    <p>Units and objects that can have ACE Arsenal added to them will be called "Boxes" in this documentation.</p>
+</div>
+
+To quickly add a full ACE Arsenal to a box for all clients (do not use in init fields):
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: hm... I'd remove the `(do not use in init fields)`
It's confusing

File: docs/wiki/framework/arsenal-framework.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+</div>
+
+<div class="panel callout">
+    <h5>Note:</h5>
+    <p>Units and objects that can have ACE Arsenal added to them will be called "Boxes" in this documentation.</p>
+</div>
+
+To quickly add a full ACE Arsenal to a box for all clients (do not use in init fields):  
+`[_box, true] call ace_arsenal_fnc_initBox`  
+`_box` being the object you wish to add ACE Arsenal to.
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: make the whole thing a code block (three backticks) and add this as a comment after the line with initBox.
You could also add another comment saying something like

> or `this` when called from the box's init field

File: docs/wiki/framework/arsenal-framework.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+Passing an empty array or `false` will still add an interaction but no additional virtual items will be added.  
+Please note that at least one virtual item need to be added otherwise ACE Arsenal will not open.
+
+if you wish to open a full ACE Arsenal on yourself or open ACE Arsenal via a custom action you can use `ace_arsenal_fnc_openBox`.
+
+`ace_arsenal_fnc_openBox`
+```cpp
+ * 0: Box <OBJECT>
+ * 1: Unit to open ACE Arsenal on <OBJECT>
+ * 2: Ignore virtual items and fill ACE Arsenal <BOOL>
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: optional

File: docs/wiki/framework/arsenal-framework.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+This will add the virtual items passed as arguments and add an ACE interaction to open ACE Arsenal.
+
+Examples:
+
+- `[_box, ["MyItem1", "MyItem2", "MyItemN"]] call ace_arsenal_fnc_initBox`
+- `[_box, true] call ace_arsenal_fnc_initBox`
+- `[_box, false, false] call ace_arsenal_fnc_initBox`
+
+Passing an empty array or `false` will still add an interaction but no additional virtual items will be added.  
+Please note that at least one virtual item need to be added otherwise ACE Arsenal will not open.
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: > one virtual item **needs** to be

File: docs/wiki/framework/arsenal-framework.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+
+### 1.2 Removing virtual items
+
+```cpp
+ * 0: Box <OBJECT>
+ * 1: Items <ARRAY of strings> <BOOL>
+ * 2: Add globally <BOOL>
+ ```
+
+Like adding virtual items, passing an array of string (classnames) will remove each ones of those items, however passing true will remove all virtual items and also remove the interaction to access ACE Arsenal.
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: > class names

File: docs/wiki/framework/arsenal-framework.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+
+`ace_arsenal_fnc_addVirtualItems`
+```cpp
+ * Arguments:
+ * 0: Box <OBJECT>
+ * 1: Items <ARRAY of strings> or <BOOL>
+ * 2: Add globally <BOOL>
+```
+
+Passing an array of strings (classnames) will add each one of those items to the specified box, passing true will add ALL items that are compatible with ACE Arsenal (that sorting is done on game startup).
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: > class names

> the sorting

> game startup? or rather mission startup?

File: docs/wiki/framework/arsenal-framework.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+
+## 2. Access to ACE Arsenal
+
+### 2.1 Adding ACE Arsenal to a box
+
+`ace_arsenal_fnc_initBox`
+```cpp
+ * 0: Box <OBJECT>
+ * 1: Items <BOOL> or <ARRAY>
+ * 2: Initialize globally <BOOL>
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: optional

File: docs/wiki/framework/arsenal-framework.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+
+- `[_box, ["item1", "item2", "itemN"]] call ace_arsenal_fnc_addVirtualItems`
+- `[_box, true, false] call ace_arsenal_fnc_addVirtualItems`
+
+### 1.2 Removing virtual items
+
+```cpp
+ * 0: Box <OBJECT>
+ * 1: Items <ARRAY of strings> <BOOL>
+ * 2: Add globally <BOOL>
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: optional

File: docs/wiki/framework/arsenal-framework.md [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+## 1. Virtual items
+
+### 1.1 Adding virtual items
+
+`ace_arsenal_fnc_addVirtualItems`
+```cpp
+ * Arguments:
+ * 0: Box <OBJECT>
+ * 1: Items <ARRAY of strings> or <BOOL>
+ * 2: Add globally <BOOL>
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: optional

File: docs/wiki/framework/arsenal-framework.md [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+ * 1: Unit to open ACE Arsenal on <OBJECT>
+ * 2: Ignore virtual items and fill ACE Arsenal <BOOL>
+```
+
+Examples:
+
+- `[_box, player] call ace_arsenal_fnc_openBox`
+- `[player, player, true] call ace_arsenal_fnc_openBox`
+
+In the second example a full ACE Arsenal will be opened on the player.
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: and in the first?

File: docs/wiki/framework/arsenal-framework.md [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+```
+
+Example:
+`[_box, true] call ace_arsenal_fnc_removeBox`
+
+## 3. Config entries
+
+ACE arsenal has 2 new config entries for items:
+
+- `ace_arsenal_hide`: `0`(shown) or `1` (hidden), used to hide items from ACE Arsenal.
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: isn't it just the other way round with vanilla? :D

File: docs/wiki/framework/arsenal-framework.md [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+ * 0: Box <OBJECT>
+ * 1: Remove globally <BOOL>
+```
+
+Example:
+`[_box, true] call ace_arsenal_fnc_removeBox`
+
+## 3. Config entries
+
+ACE arsenal has 2 new config entries for items:
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: > Arsenal

File: docs/wiki/framework/arsenal-framework.md [hunk 17]
```diff
@@ -0,0 +1,125 @@
...
+Examples:
+
+- `[_box, player] call ace_arsenal_fnc_openBox`
+- `[player, player, true] call ace_arsenal_fnc_openBox`
+
+In the second example a full ACE Arsenal will be opened on the player.
+
+### 2.2 Removing the ACE Arsenal interaction
+
+In order to remove the ACE interaction added by initBox you need to either use removeBox or remove all virtual items, since we saw how to remove all virtual items above we'll be focusing on removeBox.
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: maybe add the fully qualified function names here?

add a linkt to "how to remove all virtual items above" to the appropriate section

File: docs/wiki/framework/arsenal-framework.md [hunk 18]
```diff
@@ -0,0 +1,125 @@
...
+
+Example:
+`[_box, true] call ace_arsenal_fnc_removeBox`
+
+## 3. Config entries
+
+ACE arsenal has 2 new config entries for items:
+
+- `ace_arsenal_hide`: `0`(shown) or `1` (hidden), used to hide items from ACE Arsenal.
+- `ace_arsenal_uniqueBase`: Classname of the item that ACE Arsenal will replace it with when saving a loadout.
```
Comment (bux) [inline docs/wiki/framework/arsenal-framework.md]: > Class name

File: docs/wiki/framework/arsenal-framework.md (hunk 10 above)
Comment (alganthe) [reply docs/wiki/framework/arsenal-framework.md]: game startup, it's done on preStart

File: docs/wiki/framework/arsenal-framework.md (hunk 15 above)
Comment (alganthe) [reply docs/wiki/framework/arsenal-framework.md]: afaik:
0 false
1 true
vanilla is "curatorScope" and doesn't serve the same purpose

File: docs/wiki/framework/arsenal-framework.md (hunk 14 above)
Comment (alganthe) [reply docs/wiki/framework/arsenal-framework.md]: No, in the first the "box" virtual items would be used, in the second example the box is the player and the item list is retrieved from the cache.

File: docs/wiki/framework/arsenal-framework.md (hunk 6 above)
Comment (alganthe) [reply docs/wiki/framework/arsenal-framework.md]: is this what you're suggesting?:

```cpp
[_box, true] call ace_arsenal_fnc_initBox;
// To quickly add a full ACE Arsenal to a box for all clients
```

`_box` being the object you wish to add ACE Arsenal to. (or `this` when called from the box's init field)

File: docs/wiki/framework/arsenal-framework.md (hunk 14 above)
Comment (bux) [reply docs/wiki/framework/arsenal-framework.md]: i meant: people might want a description for the first example too

**Comment by alganthe** [comment]
@jonpas

**Pull request merged**